src/price-spread-ratio/price-spread-ratio.js

```diff
@@ -0,0 +1,166 @@
+    metadata: {
+      contractName,
+      address,
+      priceSpreadRatio,
+      lowerLimit: lower,
+      upperLimit: upper,
+      timeThreshold: threshold,
+      timeExceeded: timeDelta,
+    },
+  });
+}
+
+function provideInitialize(data) {
+  return async function initialize() {
+    // get configuration data specific to this agent
+    /* eslint-disable no-param-reassign */
+    data.config = config.priceSpreadRatio;
+
+    // initialize the block timestamp
+    data.blockTimestamp = 0;
+
+    // create an ethers provider for calling contract read-only methods
+    data.provider = new ethers.providers.JsonRpcProvider(getJsonRpcUrl());
+
+    // get the contract names that we are interested in
+    data.contractNames = Object.keys(data.config);
+
+    // load the array of contract addresses, abis, and ethers interfaces
+    data.contractsData = common.getContractAddressesAbis(data.contractNames);
+
+    // create an ethers contract object for each contract and store with the other contract data
+    // also extract the FTX url corresponding to each address and store it with the contract data
+    data.contractsData.forEach((contractData) => {
+      // create and store the ethers contract
+      contractData.contract = new ethers.Contract(
+        contractData.address,
+        contractData.contractAbi,
+        data.provider,
+      );
+
+      // get the FTX url and store it
+      contractData.ftxUrl = data.config[contractData.name].ftxUrl;
+
+      // get the upper and lower limits of the price spread ratio and store them as BigNumbers
+      const contractName = contractData.name;
+      contractData.upperLimitPercent = new BigNumber(data.config[contractName].upperLimitPercent);
+      contractData.lowerLimitPercent = new BigNumber(data.config[contractName].lowerLimitPercent);
+
+      // store the time threshold for how long the price spread ratio must be outside the limits
+      contractData.timeThresholdSeconds = BigInt(data.config[contractName].timeThresholdSeconds);
+    });
+    /* eslint-enable no-param-reassign */
+  };
+}
+
+function provideHandleBlock(data) {
+  return async function handleBlock(blockEvent) {
+    const findings = [];
+
+    const blockTimestamp = BigInt(blockEvent.block.timestamp);
+
+    // iterate over the contracts of interest to get the current prices
+    const promises = data.contractsData.map(async (contractData) => {
+      // do not create a finding the first time the handler is run
+      // instead, initialize the stored timestamp for this contract
+      if (!contractData.lastTimestampInLimits) {
+        // eslint-disable-next-line no-param-reassign
+        contractData.lastTimestampInLimits = blockTimestamp;
+        return;
+      }
+
+      const { contract } = contractData;
+
+      // get the price for each asset from the FTX
+      const ftxResponse = await axios.get(contractData.ftxUrl);
+
+      // check that we received a status 200
+      if (ftxResponse.status !== 200) {
```

Comment:
axios.get() can throw an exception

Reply:
I wrapped the `await ...` with a `try ... catch ...`, then added a check for the `success` field in the response.